worker/src/RTC/IceServer.cpp

```diff
@@ -375,17 +405,29 @@ namespace RTC
 				}
 				else
 				{
-					MS_DEBUG_TAG(ice, "transition from state 'disconnected' to 'completed'");
-
 					// Store the tuple.
 					auto* storedTuple = AddTuple(tuple);
 
-					// Mark it as selected tuple.
-					SetSelectedTuple(storedTuple);
-					// Update state.
-					this->state = IceState::COMPLETED;
-					// Notify the listener.
-					this->listener->OnIceServerCompleted(this);
+					if ((hasNomination && nomination > this->remoteNomination) || !hasNomination)
```

Comment:
Commenting here because I was re-reviewing this PR just know.

The condition previous to this `else` block assures that here there `hasUseCandidate` is `true` and `hasNominamtion` is also true, so checking that again here is not needed, and makes it harder to read. I had to go to the previous block to make sure this conditions were repeated.

This pattern repeats over this PR.

Reply:
Can you please comment this in the new ongoing PR? Otherwise I'm gonna get a bit crazy. I'll reply there, but ok, you are not right. In the else block it's not true that `hasUseCandidate` is true.